src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -888,6 +888,14 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                         if (body != null && ((methodSymbol.ContainingType.IsStructType() && !methodSymbol.IsImplicitConstructor) || instrumentForDynamicAnalysis))
                         {
+                            if (instrumentForDynamicAnalysis && methodSymbol.IsImplicitConstructor)
```

Comment:
We can assert inside the if that the instrumentForDynamicAnalysis == true.

---
In reply to: [76295380](https://github.com/dotnet/roslyn/pull/13352#discussion_r76295380) [](ancestors = 76295380)